cicd/tekton-pipelines/cluster-config/templates/deploy.yaml

```diff
@@ -0,0 +1,156 @@
+    env:
+    - name: ARGOCD_SERVER
+      value: {{ .Values.argocd.server }}
+
+  steps:
+  - name: clone-repo
+    image: alpine/git:v2.26.2
+    script: |
+      echo "Cloning the repository..."
+      git clone $(params.configGitUrl) /workspace/source
+
+  - name: update-yaml
+    image: alpine/git:v2.26.2
+    workingDir: "/workspace/source"
+    script: |
+      #!/usr/bin/env sh
+      set -e
+
+      VALUES_FILE="{{ .Values.argocd.apps.valuesPath }}"
+
+      echo "updating image tag to $(inputs.params.ImageTag)"
+      sed -i "s#$(inputs.params.IMAGE):[a-zA-Z0-9]\\+#$(inputs.params.IMAGE):$(inputs.params.ImageTag)#" "$VALUES_FILE"
+      
+      # Update the tag for the structure: tag: ""
+      sed -E -i "s#(tag:)\s*\"?[^\"]*\"?#\1 \"$(inputs.params.ImageTag)\"#" "$VALUES_FILE"
+
+  - name: commit-push-changes-gitops
+    image: alpine/git:v2.26.2
+    env:
+    - name: PARAM_USER_HOME
+      value: $(params.userHome)
+    - name: WORKSPACE_SSH_DIRECTORY_BOUND
+      value: $(workspaces.ssh-directory.bound)
+    - name: WORKSPACE_SSH_DIRECTORY_PATH
+      value: $(workspaces.ssh-directory.path)
+    - name: WORKSPACE_BASIC_AUTH_DIRECTORY_BOUND
+      value: $(workspaces.basic-auth.bound)
+    - name: WORKSPACE_BASIC_AUTH_DIRECTORY_PATH
+      value: $(workspaces.basic-auth.path)
+    - name: USER_EMAIL
+      value: $(params.userEmail)
+    workingDir: "/workspace/source"
+    script: |
+      #!/usr/bin/env sh
+      set -e
+
+      mkdir -p "${PARAM_USER_HOME}"
+
+      if [ "${WORKSPACE_BASIC_AUTH_DIRECTORY_BOUND}" = "true" ]; then
+        echo "${PARAM_USER_HOME}"
+        touch "${PARAM_USER_HOME}/.git-credentials"
+        touch "${PARAM_USER_HOME}/.gitconfig"
+        chmod 755 "${PARAM_USER_HOME}/.git-credentials"
+        chmod 755 "${PARAM_USER_HOME}/.gitconfig"
+        git config --global credential.helper cache
+        git config --global credential.helper "cache --timeout=3600"
+        export USERNAME=`cat /workspace/basic-auth/username`
+        export TOKEN=`cat /workspace/basic-auth/password`
+        git config --global user.name "${USERNAME}"
+        git config --global credential.helper "store --file ${PARAM_USER_HOME}/.git-credentials"
+        echo "https://${USERNAME}:${TOKEN}@github.com" > "${PARAM_USER_HOME}/.git-credentials"
+        chmod 400 "${PARAM_USER_HOME}/.git-credentials"
+        chmod 400 "${PARAM_USER_HOME}/.gitconfig"
+      fi
+
+      if [ "${WORKSPACE_SSH_DIRECTORY_BOUND}" = "true" ]; then
+        cp -R "${WORKSPACE_SSH_DIRECTORY_PATH}" "${PARAM_USER_HOME}/.ssh"
+        chmod 700 "${PARAM_USER_HOME}/.ssh"
+        chmod -R 400 "${PARAM_USER_HOME}/.ssh/*"
+      fi
+      
+      export USERNAME=`cat /workspace/basic-auth/username`
+      git config --global user.email "${USER_EMAIL}"
+      git config --global user.name "${USERNAME}"  # Set your name here
+
+      git add .
+      git commit --allow-empty -m "updating imagetag to $(inputs.params.ImageTag)"
+
+      git push origin main
```

Comment:
sometimes the repo has master also has default branch? can we check which is the default branch and use?

Reply:
or can we take this branch name also from values.yaml ?

Reply:
depends on where we are pushing